spotbugs/src/main/java/edu/umd/cs/findbugs/PluginLoader.java

```diff
@@ -928,6 +928,7 @@ private void loadPluginComponents()
         // Create BugPatterns
         List<Node> bugPatternNodeList = XMLUtil.selectNodes(pluginDescriptor, "/FindbugsPlugin/BugPattern");
         for (Node bugPatternNode : bugPatternNodeList) {
+            String ruleId = bugPatternNode.valueOf("@ruleId");
```

Comment:
This `ruleId` can be an empty string, right? Then I think here we need a consideration for (old) plugins.

Consider to test your implementation with a plugin. You can create distributed files by `./gradlew assemble`. Unpack the generated .tar or .zip and [run it in your local](https://spotbugs.readthedocs.io/en/stable/running.html#quick-start).

I suggest you to use find-sec-bugs to run with, but if you need a small project for faster build, you may try [findbugs-slf4j](https://github.com/kengoTODA/findbugs-slf4j) instead. It's also possible to [create a plugin project by your own](https://github.com/spotbugs/spotbugs-archetype/tree/a3e126806686bc7f280dc1e294fbf98d21903468#how-to-use).